storage_controller/src/service.rs

```diff
@@ -5715,18 +5722,70 @@ impl Service {
         self.gate.close().await;
     }
 
+    async fn secondary_lag(
+        &self,
+        secondary: &NodeId,
+        tenant_shard_id: TenantShardId,
+    ) -> Result<u64, mgmt_api::Error> {
+        let nodes = self.inner.read().unwrap().nodes.clone();
+        let node = nodes.get(secondary).ok_or(mgmt_api::Error::ApiError(
+            StatusCode::NOT_FOUND,
+            format!("Node with id {} not found", secondary),
+        ))?;
+
+        match node
+            .with_client_retries(
+                |client| async move { client.tenant_secondary_status(tenant_shard_id).await },
+                &self.config.jwt_token,
+                1,
+                3,
+                Duration::from_millis(250),
+                &self.cancel,
+            )
+            .await
+        {
+            Some(Ok(status)) => Ok(status.bytes_total - status.bytes_downloaded),
```

Comment:
`bytes_total` is only updated when they download a heatmap, so a secondary that has a lot of work to do might not realise they have a lot of work to do until they start trying to download.

So this works as a pre-check for tenants that are definitely a long way from being able to live migrate quickly, but seeing a low 'lag' number here does not guarantee that the live migration will actually complete quickly.

Reply:
Also: secondary progress can contain zeros for a secondary location that has never done a heatmap download (e.g. a new location, or right after pageserver restart), so need an opinionated handling of that case -- probably assuming it is very laggy makes sense, because we can't trust that it will cut over quickly.

Reply:
Thanks for flagging. Addressed in https://github.com/neondatabase/neon/pull/8644/commits/69dc6466ce5f17297a2dd646b87418671b520456